pkg/schema/check.go

```diff
@@ -9,11 +9,11 @@ import (
 	"github.com/k14s/ytt/pkg/yamlmeta"
 )
 
-// CheckDocument attempts type check of `doc`.
+// CheckNode attempts type check of root node and its children.
 //
 // If `d` has "schema/type" metadata (typically attached using SetType()), `d` is checked against that schema, recursively.
 // `chk` contains all the type violations found in the check.
-func CheckDocument(doc *yamlmeta.Document) TypeCheck {
+func CheckNode(doc yamlmeta.Node) TypeCheck {
```

Comment:
HA! I reviewed #580 first and made this suggestion. I'm sure you'll point my attention back here. 😆  Good stuff.